Wikipedia/Categories/NSAttributedString+WMFHTMLForSite.m

```diff
@@ -24,18 +25,27 @@ + (NSDictionary*)wmf_defaultHTMLOptionsForSite:(MWKSite*)site {
                DTDefaultFontFamily: defaultFont.familyName,
                DTDefaultFontName: defaultFont.fontName,
                DTDefaultFontSize: @(defaultFont.pointSize),
-               DTDefaultLineHeightMultiplier: @([[NSParagraphStyle defaultParagraphStyle] lineHeightMultiple]),
                DTDefaultLinkDecoration: @NO, // disable decoration for links
-               DTDefaultLinkColor: [UIColor wmf_logoBlue],
                DTDocumentPreserveTrailingSpaces: @YES,
-               DTDefaultStyleSheet: defaultStyleSheet
+               DTDefaultStyleSheet: defaultStyleSheet,
+               DTUseiOS6Attributes: @YES
     };
+    // Reminder! Use tintColor with UILabel or UITextView to control link color!
 }
 
 - (instancetype)initWithHTMLData:(NSData*)data site:(MWKSite*)site {
-    return [self initWithHTMLData:data
-                          options:[[self class] wmf_defaultHTMLOptionsForSite:site]
-               documentAttributes:nil];
+    NSAttributedString* attrStr = [self initWithHTMLData:data
+                                                 options:[[self class] wmf_defaultHTMLOptionsForSite:site]
+                                      documentAttributes:nil];
+    attrStr = [attrStr wmf_attributedStringWithParagraphStylesAdjustments:^(NSMutableParagraphStyle* paragraphStyle){
+        /*
+           Needed because if you try adjust line spacing with DTDefaultLineHeightMultiplier
+           anything larger than 1.0 ends up adding a bunch of padding before the first
+           paragraph of text.
+         */
+        paragraphStyle.lineSpacing = 12;
```

Comment:
Ya the comment on line 42 sums it up. This way we get an attributed string with no surprise padding which lets us control padding around the text via constraint constants in a predictable way.